internal/gengapic/gengapic.go

```diff
@@ -481,9 +481,14 @@ func (g *generator) returnType(m *descriptor.MethodDescriptorProto) (string, err
 
 	// Regular return type.
 	retTyp := fmt.Sprintf("*%s.%s", outSpec.Name, outType.GetName())
-
-	// Returning a custom operation, not a GET (which would be a polling RPC), use the wrapper type.
-	if g.opts.diregapic && g.aux.customOp != nil && m.GetOutputType() == g.customOpProtoName() && info.verb != "get" {
+	wrap := g.opts.diregapic && // Generator in DIREGAPIC mode.
+		g.aux.customOp != nil && // API Defines a custom operation.
+		m.GetOutputType() == g.customOpProtoName() && // Method returns the custom operation.
+		info.verb != "get" && // Method is not a GET (polling methods).
+		m.GetName() != "Wait" // Method is not a Wait (uses POST).
```

Comment:
This looks right, but let's have @vam-google confirm. These are just heuristics that will be replaced by more robust logic once the annotations to the synthetic protos get committed, right?

Reply:
That's a great question. @vam-google please confirm the heuristics I've documented here and Victor's question regarding their temporary nature.